components/Common/CodeBox/index.tsx

```diff
@@ -0,0 +1,79 @@
+import {
+  DocumentDuplicateIcon,
+  CodeBracketIcon,
+} from '@heroicons/react/24/outline';
+import classNames from 'classnames';
+import type { FC, PropsWithChildren } from 'react';
+import { useRef } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+import Button from '@/components/Common/Button';
+import Notification from '@/components/Common/Notification';
+import { useCopyToClipboard } from '@/hooks/useCopyToClipboard';
+
+import styles from './index.module.css';
+
+const languageMap: Record<string, string[]> = {
+  JSX: ['jsx'],
+  JSON: ['json'],
+  XML: ['xml'],
+  YML: ['yml', 'yaml'],
+  Bash: ['bash', 'sh', 'shell', 'zsh', 'shellscript'],
+  Shell: ['console', 'shellsession'],
+  TypeScript: ['ts', 'typescript'],
+  JavaScript: ['cjs', 'mjs', 'js', 'javascript'],
+};
+
+const labelsMap = Object.fromEntries(
+  Object.entries(languageMap).flatMap(([label, languages]) =>
+    languages.map(language => [language, label])
+  )
+);
+
+type CodeBoxProps = {
+  language: string;
+  showLineNumbers?: boolean;
+};
+
+const CodeBox: FC<PropsWithChildren<CodeBoxProps>> = props => {
+  const { children, language, showLineNumbers = false } = props;
+
+  const ref = useRef<HTMLPreElement>(null);
+
+  const [copied, copyToClipboard] = useCopyToClipboard();
+
+  const onCopy = () => {
+    if (ref.current !== null && ref.current.textContent !== null) {
```

Comment:
Hmmm, what are the cases whe it is null or textNull is null?

Reply:
This is basically for type safety.

 `ref.current ` is null when the component is not mounted and `ref.current.textContent` is null for specific elements like the `document` itself.

Reply:
Right can't you just `if(ref.current?.textContent)`